tasks/deploy/github_actions_tools.py

```diff
@@ -0,0 +1,149 @@
+import sys
+import tempfile
+import zipfile
+from datetime import datetime
+from time import sleep
+
+from invoke.exceptions import Exit
+
+from .color import color_message
+from .github import Github, GithubException
+
+
+def trigger_macos_workflow(
+    github_action_ref="master",
+    datadog_agent_ref="master",
+    release_version="nightly-a7",
+    major_version="7",
+    python_runtimes="3",
+):
+    """
+    Trigger a workflow to build a MacOS Agent.
+    """
+    inputs = {}
+
+    if datadog_agent_ref is not None:
+        inputs["datadog_agent_ref"] = datadog_agent_ref
+
+    if release_version is not None:
+        inputs["release_version"] = release_version
+
+    if major_version is not None:
+        inputs["agent_major_version"] = major_version
+
+    if python_runtimes is not None:
+        inputs["python_runtimes"] = python_runtimes
+
+    print(
+        "Creating workflow on datadog-agent-macos-build on commit {} with args:\n{}".format(
+            github_action_ref, "\n".join(["  - {}: {}".format(k, inputs[k]) for k in inputs])
+        )
+    )
+
+    # Hack: get current time to only fetch workflows that started after now.
+    now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # The workflow trigger endpoint doesn't return anything. You need to fetch the workflow run id
+    # by yourself.
+    Github().trigger_workflow("DataDog/datadog-agent-macos-build", "macos.yaml", github_action_ref, inputs)
+
+    # Thus the following hack: query the latest run for ref, wait until we get a non-completed run
+    # that started after we triggered the workflow.
+    # In practice, this should almost never be a problem, even if the Agent 6 and 7 jobs run at the
+    # same time, given that these two jobs will target different github_action_ref on RCs / releases.
+    retries = 1
+    MAX_RETRIES = 10
+    while retries <= MAX_RETRIES:
```

Comment:
Why not using a `for` loop like you did on `make_request`?

Reply:
These were made at very different times during development, I'll change it